config/src/lib.rs

```diff
@@ -66,7 +66,7 @@ pub struct Configuration {
 }
 
 /// Represents an individual config option that the program can interpret
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Ord, PartialOrd)]
```

Comment:
`Ord` and `PartialOrd` are derived and therefore all fields are considered while `PartialEq` is implemented to use only the `name` field. That leads to inconsistencies which could cause problems in the future.

Also the [Rust documentation](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html) states:
> Implementations of PartialEq, PartialOrd, and Ord must agree with each other. It's easy to accidentally make them disagree by deriving some of the traits and manually implementing others.

I think that it's necessary to also implement the traits `Ord` and `PartialOrd`.

Reply:
You are completely correct of course. I actually read that sentence in the documentation 3 minutes before writing the code, but apparently wasn't fully awake :(

I've pushed a new commit.